src/models/workspaces.ts

```diff
@@ -85,39 +86,80 @@ export const LearningLogWorkspaceModel = types
+  .model("PublishedWorkspace", {
+    type: "published",
+    document: DocumentModel,
+    createdAt: types.number,
+    userId: types.string,
+    groupId: types.string,
+    tool: types.maybe(types.undefined),
+    sectionId: types.string,
+    groupUserConnections: types.map(types.boolean),
+    mode: "1-up"
+  });
+
 export const WorkspacesModel = types
   .model("Workspaces", {
     sections: types.array(SectionWorkspaceModel),
-    learningLogs: types.array(LearningLogWorkspaceModel)
+    learningLogs: types.array(LearningLogWorkspaceModel),
+    publications: types.array(PublishedWorkspaceModel)
   })
-  .actions((self) => {
+  .views((self) => {
     const findByDocumentId = (documentKey: string) => {
       return (workspace: WorkspaceModelType) => workspace.document.key === documentKey;
     };
-    const getSectionWorkspace = (sectionId: string) => {
-      return self.sections.find((workspace) => workspace.sectionId === sectionId);
-    };
-    const getLearningLogWorkspace = (documentKey: string) => {
-      return self.learningLogs.find(findByDocumentId(documentKey));
-    };
-    const getWorkspace = (documentKey: string) => {
-      return self.sections.find(findByDocumentId(documentKey)) || self.learningLogs.find(findByDocumentId(documentKey));
-    };
-
     return {
-      getSectionWorkspace,
-      getLearningLogWorkspace,
-      getWorkspace,
+      findByDocumentId,
+
+      // Returns the most recently published docs for the given section, sorted by group ID number
+      getLatestPublicationsForSection(sectionId: string) {
+        const latestPublications: PublishedWorkspaceModelType[] = [];
+        self.publications
+          .filter((publication) => publication.sectionId === sectionId)
+          .forEach((publication) => {
+            if (publication.sectionId !== sectionId) {
+              return;
+            }
+            const groupId = publication.groupId;
+            const latestIndex = latestPublications.findIndex((pub) => pub.groupId === groupId);
+            if (latestIndex === -1) {
+              latestPublications.push(publication);
+            } else if (publication.createdAt > latestPublications[latestIndex].createdAt) {
+              latestPublications[latestIndex] = publication;
+            }
+          });
+
+        return latestPublications
+          .sort((pub1, pub2) => parseInt(pub1.groupId, 10) - parseInt(pub2.groupId, 10));
+      },
 
+      getSectionWorkspace(sectionId: string) {
+        return self.sections.find((workspace) => workspace.sectionId === sectionId);
+      },
+
+      getWorkspace(documentKey: string) {
+        return self.sections.find(findByDocumentId(documentKey))
+               || self.learningLogs.find(findByDocumentId(documentKey))
+               || self.publications.find(findByDocumentId(documentKey));
+      },
+
+      getLearningLogWorkspace(documentKey: string) {
+        return self.learningLogs.find(findByDocumentId(documentKey));
+      },
```

Comment:
We should make sure that an appropriate polyfill is in place for support of browsers without `Array.find`.